File: src/main/java/ru/spb/sspk/ssdmd/phonebook_test/bot/TelegramBot.java
package ru.spb.sspk.ssdmd.phonebook_test.bot;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;
import org.telegram.telegrambots.bots.TelegramLongPollingBot;
import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
import org.telegram.telegrambots.meta.api.objects.Update;
import org.telegram.telegrambots.meta.exceptions.TelegramApiException;
import ru.spb.sspk.ssdmd.phonebook_test.service.PersonService;
import ru.spb.sspk.ssdmd.phonebook_test.service.UserService;

@Component
@Slf4j
public class TelegramBot extends TelegramLongPollingBot {

    private final InlineKeyboardMaker inlineKeyboardMaker;
    private final UserService userService;
    private final SendMessageBot sendMessageBot;

    @Value("${telegram.name}")
    private String botName;
    @Value("${telegram.token}")
    private String botToken;
    String code = "TheProfessionalSoftwareMaintenanceDepartmentInTheBestAndJAVAIsTheBestProgrammingLanguage";

    public TelegramBot(PersonService personService, InlineKeyboardMaker inlineKeyboardMaker, UserService userService, SendMessageBot sendMessageBot) {
        this.inlineKeyboardMaker = inlineKeyboardMaker;
        this.userService = userService;
        this.sendMessageBot = sendMessageBot;
    }

    @Override
    public String getBotUsername() {
        return botName;
    }

    @Override
    public String getBotToken() {
        return botToken;
    }

    @Override
    public void onUpdateReceived(Update update) {
        if (update.hasMessage() && update.getMessage().hasText()) {
            String answer = update.getMessage().getText();
            Long userId = update.getMessage().getFrom().getId();
            Long chatId = update.getMessage().getChatId();
            try {
                SendMessage message = getCommandResponse(answer, userId);
                message.setChatId(chatId);
                execute(message);
            } catch (TelegramApiException e) {
                log.error("", e);
            }
        } else if (update.hasCallbackQuery()) {

            try {
                SendMessage message = getCommandResponse(update.getCallbackQuery().getData(),
                        update.getCallbackQuery().getFrom().getId());
                message.setChatId(update.getCallbackQuery().getMessage().getChatId());
                execute(message);
            } catch (TelegramApiException e) {
                log.error("", e);
            }
        }

    }

    SendMessage getCommandResponse(String answer, Long userId) {
        if (userId.equals(userService.findAll(userId)) &&
                answer.equals(EnumCommandBot.INFO.getCommand())) {
            return sendMessageBot.handleInfoCommand(answer);
        }

        if (userId.equals(userService.findAll(userId)) &&
                answer.equals(EnumCommandBot.FIND.getCommand())) {
            return sendMessageBot.handleFindCommand(answer);
        }

        if (userId.equals(userService.findAll(userId)) &&
                answer.equals(EnumCommandBot.CREATE.getCommand())) {
            return sendMessageBot.handleCreateCommand(answer);
        }

        if (userId.equals(userService.findAll(userId)) &&
                answer.equals(EnumCommandBot.UPDATE.getCommand())) {
            return sendMessageBot.handleUpdateCommand(answer);
        }

        if (userId.equals(userService.findAll(userId)) &&
                answer.equals(EnumCommandBot.DELETE.getCommand())) {
            return sendMessageBot.handleDeleteCommand(answer);
        }

        if (userId.equals(userService.findAll(userId)) &&
                answer.equals(EnumCommandBot.NOTFOUND.getCommand())) {
            return sendMessageBot.handleNotFoundCommand();
        }

        if (answer.equals(EnumCommandBot.START.getCommand())) {
            return sendMessageBot.handleStartCommand();
        }

        if (answer.equals(code)){
            return sendMessageBot.handleSaveUserBot(userId);
        }

        return sendMessageBot.handleStandardCommand(answer, userId);
    }

}

File: src/main/java/ru/spb/sspk/ssdmd/phonebook_test/bot/TelegramBot.kt
package ru.spb.sspk.ssdmd.phonebook_test.bot

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component
import org.telegram.telegrambots.bots.TelegramLongPollingBot
import org.telegram.telegrambots.meta.api.methods.send.SendMessage
import org.telegram.telegrambots.meta.api.objects.Update
import org.telegram.telegrambots.meta.exceptions.TelegramApiException
import ru.spb.sspk.ssdmd.phonebook_test.service.UserServiceImpl

@Component
class TelegramBot(
    @Value("\${telegram.token}")
    private val botToken: String,
    @Value("\${telegram.name}")
    private val botUsername: String,
    private val userService: UserServiceImpl,
    private val sendMessageBot: SendMessageBot,
    private val logger: Logger = LoggerFactory.getLogger(TelegramBot::class.java)
) : TelegramLongPollingBot(botToken) {

    override fun getBotUsername(): String {
        return botUsername
    }

    override fun onUpdateReceived(update: Update) {
        if (update.hasMessage()) {
            val answer: String = update.message.text
            val userId: Long = update.message.from.id
            val userName: String = update.message.from.userName
            val chatId: Long = update.message.chatId
            try {
                val message = getCommandResponse(answer, userId, userName)
                message.setChatId(chatId)
                execute(message)
                logger.info("Write message: user=$userId, $userName, '$answer'")
            } catch (e: TelegramApiException) {
                logger.error("", e)
            }
        } else if (update.hasCallbackQuery()) {
            val queryData: String = update.callbackQuery.data
            val userId: Long = update.callbackQuery.from.id
            val userName: String = update.callbackQuery.from.userName
            val chatId: Long = update.callbackQuery.message.chatId
            logger.info("InlineKeyBoard click: user=$userId, $userName, $queryData")
            try {
                val message = getCommandResponse(queryData, userId, userName)
                message.setChatId(chatId)
                execute(message)
            } catch (e: TelegramApiException) {
                logger.error("", e)
            }
        }
    }

    private fun getCommandResponse(answer: String, userId: Long, username: String): SendMessage {

        return if (userService.isUserAuthentication(userId) && answer.isNotEmpty()) {
            if (userService.checkingUserRole(userId) && answer == "/users") sendMessageBot.showAllUsersForAdmin()
            else {
                when (answer) {
                    "/start" -> sendMessageBot.handleStartCommand(userId, username)
                    "/help" -> sendMessageBot.handleHelpCommand(userId)
                    "/users" -> sendMessageBot.executionOfAccessDenied(userId, username)
                    else -> sendMessageBot.handleStandardCommand(userId, answer)
                }
            }
        } else if (userService.checkingUserPresence(userId) && answer.isNotEmpty()) {
            when (answer) {
                "/start" -> sendMessageBot.handleStartCommand(userId, username)
                "/help" -> sendMessageBot.handleHelpCommandNotAuthentication()
                "/users" -> sendMessageBot.executionOfAccessDenied(userId, username)
                else -> sendMessageBot.handleLackOfAccessOrAuthentication(userId, username, answer)
            }
        } else if (!userService.checkingUserPresence(userId)) {
            when {
                answer == "/start" -> sendMessageBot.handleStartCommand(userId, username)
                answer == "/help" -> sendMessageBot.handleHelpCommandNotAuthentication()
                answer.isNotEmpty() -> sendMessageBot.executionOfAccessDenied(userId, username)
                else -> sendMessageBot.handleNotFound()
            }
        } else sendMessageBot.handleNotFound()
    }
}
